Sources/GRPCCore/Transport/ServerTransport.swift

```diff
@@ -15,21 +15,37 @@
  */
 
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+/// A protocol server transport implementations must conform to.
 public protocol ServerTransport: Sendable {
   typealias Inbound = RPCAsyncSequence<RPCRequestPart>
   typealias Outbound = RPCWriter<RPCResponsePart>.Closable
+  typealias AcceptedStreams = RPCAsyncSequence<RPCStream<Inbound, Outbound>>
+  typealias ListenEvent = Result<AcceptedStreams, any Error>
 
-  /// Starts the transport and returns a sequence of accepted streams to handle.
+  /// A sequence of ``ListenEvent``s, describing whether the transport was started successfully or not.
+  ///
+  /// This sequence should only return a single event. If multiple events are yielded into it, they may be ignored.
+  /// Not yielding any events or finishing the sequence without yielding any events is considered a broken
+  /// implementation of this protocol.
+  ///
+  /// Once ``listen()`` stops running, the sequence should be finished if it hasn't been finished already.
+  var listenEventStream: RPCAsyncSequence<ListenEvent> { get }
```

Comment:
I think we should broaden this slightly to be "transport events". That gives us a bit of scope in the future to extend what a transport can do. We can probably just call the property "events" and the type "TransportEvent".

Reply:
Okay, this makes sense. Should I turn the `TransportEvent` into its own type instead of being a `typealias` then?
That's what I had originally done but I turned it into an alias of `Result` because it was basically the same and felt unnecessary. But if we want it to be extendable maybe we want it to be an actual type?